keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -241,11 +240,22 @@ nitty-gritty.
   degree of mutual trust and shared ownership that share services amongst
   themselves. Membership in a supercluster is symmetric and transitive. The set
   of member clusters are mutually aware, and agree about their collective
-  association.
-- **mcsd-controller** - A controller that syncs services across clusters and
-  makes them available for multi-cluster service discovery (MCSD) and
-  connectivity. There may be multiple implementations, this doc describes
-  expected common behavior.
+  association. Within a supercluster, [namespace sameness] applies and all
+  namespaces with a given name are considered to be the same namespace.
+- **mcs-controller** - A controller that syncs services across clusters and
+  makes them available for multi-cluster service discovery and connectivity.
+  There may be multiple implementations, this doc describes expected common
+  behavior. The controller may be a single controller, multiple decentralized
+  controllers, or a human using kubectl to create resources. This document aims
+  to support any implementation that fulfills the behavioral expectations of
+  this API.
+- **cluster name** - A unique name or identifier for the cluster, scoped to the
+  implemenetation's cluster registry. We do not attempt to define the registry.
+  Each cluster must have a name that can uniquely identify it within the
+  supercluster. A cluster name must be a valid [RFC
```

Comment:
I am skeptical that we need to define the format of a cluster name, or rather this overlaps a lot with ClusterID, which is not resolved.

Reply:
Agreed this is probably a bigger conversation and would be better solved with Cluster ID.

If we intend to use the cluster name as part of the headless implementation we need to specify some constraints here, but this does seem like a good time to bring back Cluster ID

Reply:
We decided not to define further yet - but that'll be a good thing to keep in mind when we do. uid of kube-system is at least valid as a cluster name for MCS purposes.

Reply:
The cluster name will be used in the per-pod records for headless services. That drives the reference to RFC1123. I think we can solve uniqueness issues for dependent resources in other ways. For resources like a lease that might use the name, we can e.g. add a well known prefix to avoid collisions with non-MCS-cluster named resources. I don't think we need to spec that. The DNS label requirement seems like the minimum restriction we can get away with.

Reply:
ACK.  The DNS use-case is compelling